src/scipp/core/data_group.py

```diff
@@ -248,6 +240,21 @@ def _data_group_binary(func: Callable, dg1: DataGroup, dg2: DataGroup, *args,
     })
 
 
+def _data_group_inplace(op: str, dg: DataGroup,
+                        other: Union[DataArray, Variable, numbers.Real]) -> DataGroup:
+    if isinstance(other, DataGroup):
+        if extra_keys := other.keys() - dg.keys():
+            raise ValueError(
+                "Cannot apply inplace operation because the right hand "
+                f"side has keys that are not in the left hand side: {extra_keys}")
+        for key in dg.keys() & other.keys():
+            getattr(dg[key], op)(other[key])
```

Comment:
If any of these fails, we will have partially modified `dg`. Can we do more thorough checks, so at least the risk is reduced? `Dataset` goes into quite a bit of effort to do this, iirc.

Reply:
Do you mean checking that types, units, and sizes are compatible? I don't know how to do that in Python without huge effort.

Reply:
Yes, that is what I meant. `Dataset` and `DataArray` go even further, and check that coords and bin sizes are compatible.

I am not claiming that this is the only choice, but we have to be clear about it. We can do that separately if you prefer.

Reply:
If we do not prevent such problems now, I feel at the very least we have to make it clear in the error message by including something like this:

```python
Operation fail. Parts of the data group were modified.
- Modified items: ['a', 'b']
- Unmodified items: ['d', 'e']
- Failed item (unknown state, may be modified or corrupted): 'c' 
```

Reply:
Looking at the dry run functions, it looks like data group would need extra implementations for dataset and nested data groups. And then there are other Python types which might support in place operations, e.g. lists. Making sure that  nothing gets modified in an in-place operation seems pretty difficult.
This made me think, is there much value in even supporting these operations in general? Should we limit them to specific item types or maybe not support them at all?

Reply:
You mean all the `__i*__` ones? I think I agree, it is hard to get right (without too much code), so we should maybe not support them for now? We can always to that later, in case it is really required.

Reply:
Done. Left a comment for when we ask ourselves in the future why we didn't implement them.